spec/hotel_ledger_spec.rb

```diff
@@ -0,0 +1,146 @@
+require_relative "spec_helper"
+require "date"
+
+describe "HotelLedger class" do
+  it "must list all the reservations" do
+    reservations = Hotel::HotelLedger.new.reservations
+    expect(reservations).must_equal []
+  end
+
+  it "must list all rooms in the hotel" do
+    rooms = Hotel::HotelLedger.new.rooms
+    expect(rooms).must_equal [*(1..20)]
+  end
+
+  it "can reserve a room for a given date range" do
```

Comment:
You should also verify that you **can** reserve a room starting on the same date the previous reservation ended.